plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -196,6 +205,7 @@ final class AzureStorageSettings {
     private final String account;
     private final String connectString;
     private final String endpointSuffix;
+    private final String tokenCredentialType;
```

Comment:
We able to pass the specific enum class instead of string here? Like with LocationMode? With so many strings next to each other like this using the intended class would reduce changes of accidentally swapping parameters.